components/camel-core-osgi/src/main/java/org/apache/camel/core/osgi/OsgiServiceRegistry.java

```diff
@@ -62,18 +67,29 @@ public OsgiServiceRegistry(BundleContext bc) {
     }
 
     /**
-     * It's only support to look up the ServiceReference with Class name
+     * It's only support to look up the ServiceReference with Class name or service PID
      */
     public Object lookupByName(String name) {
         Object service = null;
-        if (service == null) {
-            ServiceReference<?> sr = bundleContext.getServiceReference(name);            
-            if (sr != null) {
-                // Need to keep the track of Service
-                // and call ungetService when the camel context is closed 
-                serviceReferenceQueue.add(sr);
-                service = bundleContext.getService(sr);
-            } 
+        ServiceReference<?> sr = bundleContext.getServiceReference(name);
+        if (sr != null) {
+            // Need to keep the track of Service
+            // and call ungetService when the camel context is closed 
+            serviceReferenceQueue.add(sr);
+            service = bundleContext.getService(sr);
+        } else {
+            // trying to lookup service by PID if not found by name
+            try {
+                ServiceReference<?>[] refs = bundleContext.getServiceReferences((String)null, "(" + Constants.SERVICE_PID + "=" + name + ")");
+                if (refs != null && refs.length > 0) {
+                    // just return the first one
+                    sr = refs[0];
+                    serviceReferenceQueue.add(sr);
+                    service = bundleContext.getService(sr);
+                }
+            } catch (InvalidSyntaxException ex) {
```

Comment:
We need to do something bette than log an ERROR. Either its a DEBUG/TRACE logging error to ignore this. Or its something that needs to be wrapped and thrown as runtime exception. And is there any API on OSGi you can use to check if the syntax is correct and then just do a if correct then ... else ignore,